packages/test-tools/src/mockedEventStore.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Aggregate,
+  EventDetail,
+  EventStore,
+  EventType,
+  EventTypesDetails,
+} from '@castore/core';
+import { InMemoryStorageAdapter } from '@castore/inmemory-event-storage-adapter';
+
+export class MockedEventStore<
+  I extends string = string,
+  E extends EventType[] = EventType[],
+  $D extends EventDetail = EventTypesDetails<E>,
+  // cf https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-rc/#optional-variance-annotations-for-type-parameters
```

Comment:
Peut-être juste `// see original EventStore implementation for more details`, comme ça on évite de dupliquer deux commentaires